### PR TITLE
add runtime error handling for POSIX.

### PR DESCRIPTION
**Pull request by rgrover**
@0xc0170 @bogdanm

### CODE REVIEW

File: source/core-util_mbed_os.c [hunk 1]
```diff
@@ -48,5 +48,30 @@ void core_util_assert_internal(const char *expr, const char *file, int line, con
     mbed_die();
 }
 
-#endif // #ifdef TARGET_LIKE_MBED
+#elif defined(TARGET_LIKE_POSIX)
```
Comment (bogdanm) [inline source/core-util_mbed_os.c]: The idea behind the `core-util_mbed_os.c` file name was that it was implementing the required `core-util` functions only for mbed OS platforms. For POSIX, I'd create a new `core-util_posix.c` file.

**Comment by rgrover** [comment]
@bogdanm @0xc0170

**Comment by bogdanm** [comment]
+1

**Comment by 0xc0170** [comment]
+1

**Comment by 0xc0170** [comment]
I am looking at what we have on master at the moment, only new additions , we can release a new patch version.

**Comment by bogdanm** [comment]
Yes, patch will be fine.

**Pull request merged**